python_modules/libraries/dagster-aws/dagster_aws/ecs/utils.py

```diff
@@ -1,6 +1,55 @@
 import re
+from typing import Any, Dict
+
+from .tasks import DagsterEcsTaskDefinitionConfig
 
 
 def sanitize_family(family):
     # Trim the location name and remove special characters
     return re.sub(r"[^\w^\-]", "", family)[:255]
+
+
+def task_definitions_match(
+    desired_task_definition_config: DagsterEcsTaskDefinitionConfig,
+    existing_task_definition: Dict[str, Any],
+    container_name: str,
+) -> bool:
+    if not any(
+        [
+            container["name"] == container_name
+            for container in existing_task_definition["containerDefinitions"]
+        ]
+    ):
+        return False
+
+    existing_task_definition_config = DagsterEcsTaskDefinitionConfig.from_task_definition_dict(
+        existing_task_definition, container_name
+    )
+
+    # Exclude command since it will be overridden anyway, sidecars are checked below
+    match_without_sidecars = existing_task_definition_config._replace(
+        sidecars=[],
+    ) == desired_task_definition_config._replace(
+        sidecars=[],
+    )
+    if not match_without_sidecars:
+        return False
+
+    # Just match sidecars on certain fields
+    return [
```

Comment:
Nit: maybe switch to `if (test): return False` like the other ones for consistency. And then explicitly` return True` if we reach the end. Makes it easier to scan for what each equality test is doing.